src/index.ts

```diff
@@ -36,12 +37,21 @@ async function Main() {
 
   const userStore = new UserStore();
   const challengeStore = new ChallengeStore();
+  const checkIgnore = function(member: GuildMember | null) {
+    return (member?.roles.cache.find((r) => r.id === settingsStore.settings.role));
```

Comment:
I dont think this function would work if we changed the settings.role (I think it does it by val). it should be brought into the client.on message event so that every time we get a new message we recacluate based on the current value of settingsStore.role